### PR TITLE
Split readme into docs directory and setup sphinx

### LINKED ISSUE #74: Readme is too long and needs breaking up
Creating this so we don't forget...

Should we split the Readme into files in a doc directory in the git repository itself or should we use the wiki on GitHub? My preference is a doc directory as we're then not tied to GitHub too closely.

### PR DESCRIPTION
**Pull request by betafive**
* The top-level README document is split into multiple markdown files within a docs directory.

* Configuration for the Sphinx documentation system is created (using `sphinx-quickstart`). Documentation can now be built using `make html` or `make latexpdf` in the docs directory.

Fixes #74

### CODE REVIEW

**Comment by betafive** [comment]
Attached PDF and HTML output to show how this comes out:

[meta-raspberrypi.pdf](https://github.com/agherzan/meta-raspberrypi/files/1021903/meta-raspberrypi.pdf)

[meta-raspberrypi-html-docs.tar.gz](https://github.com/agherzan/meta-raspberrypi/files/1021908/meta-raspberrypi-html-docs.tar.gz)

File: docs/extra-build-config.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+1. Overwrite IMAGE_FSTYPES in local.conf
+  * `IMAGE_FSTYPES = "tar.bz2 ext3.xz"`
+
+2. Overwrite SDIMG_ROOTFS_TYPE in local.conf
+  * `SDIMG_ROOTFS_TYPE = "ext3.xz"`
+
+3. Overwrite SDIMG_COMPRESSION in local.conf
+  * `SDIMG_COMPRESSION = "xz"`
+
```
Comment (mirzak) [inline docs/extra-build-config.md]: Numbering is not correct when converted to pdf/hmtl. It says 1. on all of them

File: docs/extra-build-config.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+There are a set of ways in which a user can influence different paramenters of
+the build. We list here the ones that are closely related to this BSP or
+specific to it. For the rest please check:
+<http://www.yoctoproject.org/docs/latest/ref-manual/ref-manual.html>
+
+## Compressed deployed files
+
+1. Overwrite IMAGE_FSTYPES in local.conf
+  * `IMAGE_FSTYPES = "tar.bz2 ext3.xz"`
```
Comment (mirzak) [inline docs/extra-build-config.md]: These should be intended properly, currently only one space, should be 4? Intending properly also seem to fix numbering (see below comment)

1. Overwrite IMAGE_FSTYPES in local.conf
     * `IMAGE_FSTYPES = "tar.bz2 ext3.xz"`
2. Overwrite SDIMG_ROOTFS_TYPE in local.conf
    * `SDIMG_ROOTFS_TYPE = "ext3.xz"`
3. Overwrite SDIMG_COMPRESSION in local.conf
     * `SDIMG_COMPRESSION = "xz"`

File: docs/extra-apps.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Extra apps
+
+## omxplayer
+
+omxplayer depends on libav which has a commercial license. So in order to be
+able to compile omxplayer you will need to whiteflag the commercial
+licenseadding to you local.conf:
+`LICENSE_FLAGS_WHITELIST = "commercial"`
```
Comment (mirzak) [inline docs/extra-apps.md]: This looks strange to me. The output is something like this:

licenseadding to you local.conf:
`LICENSE_FLAGS_WHITELIST = "commercial"`

I think this looks better

licenseadding to you local.conf:

    LICENSE_FLAGS_WHITELIST = "commercial"

Above is a new line and indented 4 spaces. There are a lot of places across the documents where my alternative would look better and improve read-ability. IMO.

File: docs/extra-apps.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Extra apps
+
+## omxplayer
+
+omxplayer depends on libav which has a commercial license. So in order to be
+able to compile omxplayer you will need to whiteflag the commercial
+licenseadding to you local.conf:
```
Comment (mirzak) [inline docs/extra-apps.md]: This last line is a bit strange.

Maybe it should say: "So in order to be able to compile omxplayer you will need to whiteflag the commercial license in your local.conf:"

File: README.md [hunk 5]
```diff
@@ -8,283 +8,46 @@ Yocto BSP layer for the Raspberry Pi boards - http://www.raspberrypi.org/ .
...
-  - kernel_configure_variable BLK_DEV_INITRD y
-  - kernel_configure_variable INITRAMFS_SOURCE ""
-  - kernel_configure_variable RD_GZIP y
-* Set the yocto variables (in linux-raspberrypi.inc for example)
-  - `INITRAMFS_IMAGE = "<a name for your initramfs image>"`
-  - `INITRAMFS_IMAGE_BUNDLE = "1"`
-* Set the meta-rasberrypi variable (in raspberrypi.conf for example)
-  - `KERNEL_INITRAMFS = "-initramfs"`
-
-#### 3.K. Device tree support
```
Comment (mirzak) [inline README.md]: @agherzan you can correct me :).

But I believe that this section is no longer needed? All kernels that we have use device tree now days

**Comment by betafive** [comment]
@mirzak: I think readthedocs.org would be an excellent place to host this. I want to get local builds of the documentation into HTML and PDF working first though and then look at hosting.

I've not gone through the output with a fine-toothed comb so you've definitely picked up a few things I missed.

File: docs/extra-apps.md (hunk 3 above)
Comment (betafive) [inline docs/extra-apps.md]: Agreed. I've not gone through and done this kind of tidy up yet, that was going to be the next step after this initial conversion.

File: docs/extra-apps.md (hunk 4 above)
Comment (betafive) [inline docs/extra-apps.md]: Yea, there's a few editorial clean ups to be made. I'll add this to the list.

File: docs/extra-build-config.md (hunk 2 above)
Comment (betafive) [inline docs/extra-build-config.md]: Again, this tidy up and fixing the output is still todo. You've got the right fix here though.

File: docs/extra-build-config.md (hunk 1 above)
Comment (betafive) [inline docs/extra-build-config.md]: And again, agreed this is todo.

**Comment by betafive** [comment]
@mirzak: Would you like to fix the issues you've identified and add a commit to this branch? If not I'll try to have a look at it later today or Thursday.

File: README.md (hunk 5 above)
Comment (betafive) [inline README.md]: I'd agree, all non-device tree kernel versions are now gone on the master branch and pyro at least. There may be some code we can remove from the kernel recipes/classes as well, I'd need to check.

**Comment by mirzak** [comment]
> @mirzak: Would you like to fix the issues you've identified and add a commit to this branch? If not I'll try to have a look at it later today or Thursday.

I can fix it, but not sure if I can push to this branch?

**Comment by betafive** [comment]
@mirzak: If you don't have write access just use a fork of this repository and point me at the right branch, I can merge it into this.

**Comment by mirzak** [comment]
You can find the fixes here: https://github.com/mirzak/meta-raspberrypi/commits/pr/75

**Comment by betafive** [comment]
@mirzak: Looks great! Output is formatted much better as well. I'll pull that in and throw in one more commit to make sure all the links show up correctly.

**Comment by agherzan** [comment]
http://agherzan.github.io/meta-raspberrypi-docs/ Very nice!

**Comment by agherzan** [comment]
@betafive is this done? Anything else to add?

**Comment by betafive** [comment]
I think this is all done and ready to merge.

**Pull request merged**